smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/KafkaSink.java

```diff
@@ -330,12 +330,20 @@ public void isAlive(HealthReport.HealthReportBuilder builder) {
 
     public void isReady(HealthReport.HealthReportBuilder builder) {
         // This method must not be called from the event loop.
-        if (health != null) {
+        if (health != null && this.configuration.getHealthReadinessEnabled()) {
             health.isReady(builder);
         }
         // If health is disable do not add anything to the builder.
     }
 
+    public void isStarted(HealthReport.HealthReportBuilder builder) {
+        // This method must not be called from the event loop.
+        if (health != null) {
```

Comment:
Should we also have a config for enabling/disabling startup health checks?

Reply:
Hmm, probably no. We don't have one for liveness either. And the startup checks are rather basic, so we should be fine.

Reply:
In terms of config I wasn't sure, that's why this PR is a draft. I have several pending questions : 
- Do we have a config for enable/disable startup-topic-verification ? 
- A separate config for startup topic verification timeout ? 
- Or do we deprecate/change the existing `health-readiness-topic-verification` flag and `health-readiness-timeout` config options to `health-topic-verification` and `health-check-timeout` ?
I'd vote for the third one.
@Ladicek let me know what do you think

Reply:
Alright, let me try to recap all the probes and what we do. Because I'm a little lost personally :-)

### Startup

If a pod has a startup probe defined, Kubernetes will start with it and will ignore liveness and readiness (at the beginning). Once the startup probe returns "up", Kubernetes will start probing for liveness and readiness (and startup won't be checked anymore). If a pod doesn't have a startup probe, Kubernetes will start directly probing for liveness and readiness.

This is mostly useful for applications that take a long while to start, and during that time, the liveness and readiness handling doesn't make sense.

In our case, I'm not sure startup probe has a lot of meaning -- at the very most, it should just check that we have a connection to the broker. That's exactly what you're doing in this PR, so it's fine.

There could be a config option to disable startup probe, but if it only returns "down" when we're sure we didn't start yet, I think we can do without the config option.

### Liveness

If a pod has a liveness probe, Kubernetes will check it and restart the pod if liveness reports "down".

This is useful when we can tell that the application can't make progress. Restarting is usually a decent way to get out of such situation.

In our case, we know we can't make progress if a channel has failed -- because that is irreparable. That's what our liveness probe checks, so again, that's fine.

I don't think there should be a config option for disabling this. Channel failing is fatal, messages can't be processed anymore.

### Readiness

If a pod has a readiness probe, Kubernetes will check it and remove the pod from load balancer rotation if readiness reports "down".

This is useful when we can tell that the application won't be able to respond to users. Like, if all endpoints of the application require sending a message through Reactive Messaging, and we know sending such message is not gonna work, then it makes sense to mark the pod as not ready.

In our case, frankly, I'm not sure what's the value of readiness.

For a producer, we check:
- metric based: we have a connection to the broker;
- admin client based: we have a connection to the broker and the topic (to which we produce) exists on the broker.

For a consumer, we check:
- metric based: we have a connection to the broker;
- admin client based: we have a connection to the broker and we have at least 1 partition (of the topic we consume) assigned to ourselves.

I may be able to understand why we report readiness for producers, but really not sure about consumers. Especially if the number of partitions may easily be less than the number of applications in the same consumer group.

If anything, I can totally see why there must be a config option to disable readiness.

Sorry, this probably isn't very helpful, but it's my current opinion :-)

Reply:
That said, I can see that with this PR, the readiness config would affect the startup probe too, and that's not nice. So we should do something about that for sure.

Reply:
:) nevertheless it is a good recap. 
+1 for what you wrote on startup and liveness checks.
For readiness, I think there is a use case where one would like to report a pod as "not ready" if it doesn't have any partition assignments, so it won't be consuming any data and it may be that one does not want to accept traffic to that pod.
As you said it makes sense that we have an option to disable the readiness check altogether.

Reply:
> That said, I can see that with this PR, the readiness config would affect the startup probe too, and that's not nice. So we should do something about that for sure.

Yes my doubts were more for the startup probe config:
- With this PR, startup is enabled whether the readiness check is enabled or not.
- But the admin client based check is enabled by the same config `health-readiness-topic-verification` – for the draft :)
- Idem for the `health-readiness-timeout`.

I suggest that we rename those config to `health-check-topic-verification` and `health-check-timeout`. 
With the admin based check, we do essentially the same call to the broker for startup and readiness. I think it makes sense to have the same config enabling the topic verification and configure the timeout for the call.

The other option is to have separate options for startup as `health-startup-topic-verification` and `health-startup-timeout`.

Reply:
Yea I think I agree. With metric based health, startup and readiness are identical. With admin client based health, startup is a subset of readiness, which makes sense I guess.  I don't think there's a good reason to have one of them metric based and the other admin client based.

So +1 to the rename you propose.

(Alternatively, we could rename them to `health-admin-client-enabled` and `health-admin-client-timeout`. The term "topic verification" is nicely descriptive, but doesn't embody the fact that admin client is required, which is IMHO important.)

Not sure how to handle backwards compatibility though :-)

Reply:
Yes, I'm not sure "readiness" is very useful for the consumer side. 

For the producer yes. For the producer something I was thinking is the following: it should report _not ready_ if we start having issues/lag when writing to Kafka. There is a "request" protocol to determine the number of inflight records. If we reach the max, it would make sense to mark the pod as non-ready. Such behavior should be configurable. 

For consumer, it does not affect the traffic.

Reply:
About `alias`, yes, it's not for renaming but for global "shorter" properties.